projects/autocomplete-lib/src/lib/autocomplete/autocomplete.component.ts

```diff
@@ -244,17 +248,10 @@ export class AutocompleteComponent implements OnInit, OnChanges, AfterViewInit,
   public filterList() {
     this.selectedIdx = -1;
     this.initSearchHistory();
+
     if (this.query != null && this.data) {
       this.toHighlight = this.query;
-      this.filteredList = this.data.filter((item: any) => {
-        if (typeof item === 'string') {
-          // string logic, check equality of strings
-          return item.toLowerCase().indexOf(this.query.toLowerCase()) > -1;
-        } else if (typeof item === 'object' && item instanceof Object) {
-          // object logic, check property equality
-          return item[this.searchKeyword].toLowerCase().indexOf(this.query.toLowerCase()) > -1;
-        }
-      });
+      this.filteredList = this.filter !== undefined ? this.filter([...this.data], this.query) : this.defaultFilterFunction()
```

Comment:
I am (at least shallowly) destructuring `this.data` as `[...this.data]` to not let (at least for strings) `this.data` to be changed  directly as an argument when using a custom filtering as it may cause bugs